### ISSUE TITLE
Tier-1 gateway is deployed in SR mode when cluster path defined and HA_MODE set to NONE instead of DR.

### ISSUE THREAD

**Issue by davidvanr** [vmware/terraform-provider-nsxt#1086]
### Describe the bug

When creating a Tier-1 gateway with ha_mode set to none, but the edge_cluster_path is defined the ha_mode will be set to active/standby instead of distributed. 
When performing a second time the terraform apply, the mode goes from active/standby to distributed. 

It works but you need to perform 2 times the Terraform Apply. 
When not specifying the edge_cluste_path and ha_mode to none it works without issues. 

### Reproduction steps

Create a Tier-1 gateway with the ha_mode set to "NONE" and set the edge_cluster_path.

I removed the below IDs and replaced it with REMOVEDID.

```
  # nsxt_policy_tier1_gateway.tier1_gw will be created
  + resource "nsxt_policy_tier1_gateway" "tier1_gw" {
      + default_rule_logging      = true
      + description               = "Customer A Tier-1"
      + display_name              = "t1_customerA"
      + edge_cluster_path         = "/infra/sites/default/enforcement-points/default/edge-clusters/REMOVEDID"
      + enable_firewall           = false
      + enable_standby_relocation = false
      + failover_mode             = "NON_PREEMPTIVE"
      + force_whitelisting        = false
      + ha_mode                   = "NONE"
      + id                        = (known after apply)
      + ipv6_dad_profile_path     = (known after apply)
      + ipv6_ndra_profile_path    = (known after apply)
      + nsx_id                    = (known after apply)
      + path                      = (known after apply)
      + pool_allocation           = "ROUTING"
      + revision                  = (known after apply)
      + route_advertisement_types = (known after apply)
      + tier0_path                = "/infra/tier-0s/REMOVEDID"

      + context {
          + project_id = "REMOVEDID"
        }
    }

```

When applying the second Terraform apply I have the below:

```
  # nsxt_policy_tier1_gateway.tier1_gw will be updated in-place
  ~ resource "nsxt_policy_tier1_gateway" "tier1_gw" {
      ~ ha_mode                   = "ACTIVE_STANDBY" -> "NONE"
        id                        = "REMOVEDID"
        # (16 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }
```

### Expected behavior

Tier-1 ha_mode should be set to Distributed("NONE") and not to Active/passive

### Additional context

Tested in NSX version  4.1.0.2.0.21761691

**Comment by ksamoray** [comment]
Hi @davidvanr,
`ha_mode` behavior will change in the upcoming release, due to PR #1042.
The way NSX handles ha_mode has been changed in v4.1.1 and now NSX will now behave differently than in v4.1.0.

**Comment by salv-orlando** [comment]
As this issue the same issue we identified and fixed with PR #1042, I am closing this issue.
Fix will be available in the 3.5.0 release of the provider.

**Comment by salv-orlando** [comment]
Duplicate of #1042
